src/components/PageDefinitions.js

```diff
@@ -87,20 +88,35 @@ class PageDefinitions extends AbstractPageDefinitions {
     }
   }
 
+  // Get an array of definitions asynchronous, split them into 100 chunks and alert the user when they're all done
+  getDefinitionsAndNotify(definitions, message) {
+    const { dispatch, token } = this.props
+    const chunks = chunk(definitions, 100)
+    let todo = chunks.length
+    for (let i = 0; i < chunks.length; i++) {
+      dispatch(getDefinitionsAction(token, chunks[i])).then(() => {
+        if (--todo === 0) {
```

Comment:
feels like this should be more like 

```javascript
Promise.all(chunks.map(chunk => dispatch(getDefinitionAction(...))).then(()=> notify)
```

likely should have a error case where we tell the user that something bad happened. Perhaps the caller can pass in a failure message?